File: libs/backend/domain/src/main/java/reengineering/ddd/teamai/model/Message.java
package reengineering.ddd.teamai.model;

import reengineering.ddd.archtype.Entity;
import reengineering.ddd.teamai.description.MessageDescription;

public class Message implements Entity<String, MessageDescription> {
  private String identity;
  private MessageDescription description;

  public Message(String identity, MessageDescription description) {
    this.identity = identity;
    this.description = description;
  }

  public Message() {
  }

  @Override
  public String getIdentity() {
    return identity;
  }

  @Override
  public MessageDescription getDescription() {
    return description;
  }
}

File: libs/frontend/domain/src/lib/model/message.ts
import { Entity } from '../archtype/index.js';
import { MessageDescription } from '../description/index.js';

export class Message implements Entity<string, MessageDescription> {
  constructor(
    private identity: string,
    private description: MessageDescription
  ) {}

  getIdentity(): string {
    return this.identity;
  }

  getDescription(): MessageDescription {
    return this.description;
  }
}
